### PR TITLE
AF-4: Improving server structure, seedDB, connecting client and server

### PR DESCRIPTION
**Pull request by vlasichok**
Improving server structure, seedDB, connecting client and server
Next task were completed:
1. Complete MVC on server-side
2. shared -> services 
3. shared -> models
4. EVENTS const to MongoDB seed
5. Create MongoDB migration seed:run
6. Remove EVENTS const 
7. Change events.service to get data from DB

### CODE REVIEW

File: afisha-client/src/app/events/event.component.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ export class EventComponent implements OnInit {
 
   ngOnInit() {
     let id = this.route.snapshot.params['id'];
-    this.event = this.eventService.readEvent(+id);
+    this.eventService.getOneEvent(id)
+      .subscribe(
+        (event: IEvent) => this.event = event,
+        (error: any) => console.log(error),
+        () => console.log("The event with id " + id + " retrieved")
```
Comment (doslechescake) [inline afisha-client/src/app/events/event.component.ts]: In 99% cases complete is redundant. As in your case

File: afisha-client/src/app/events/events-list.component.ts [hunk 2]
```diff
@@ -15,7 +15,13 @@ export class EventsListComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.events = this.eventService.getEvents();
+    this.eventService.getEvents()
+      .subscribe(
+        (events: IEvent[]) => this.events = events,
+        (error: any) => console.log(error),
```
Comment (doslechescake) [inline afisha-client/src/app/events/events-list.component.ts]: The same like above

File: afisha-client/src/app/shared/constants/path.constant.ts [hunk 3]
```diff
@@ -6,5 +6,6 @@ export const PATH = {
   ROOT: '',
   LOGIN: 'login',
   SEARCH: 'search',
-  ABOUT: 'about'
+  ABOUT: 'about',
+  API: 'http://localhost:8800/api/'
```
Comment (doslechescake) [inline afisha-client/src/app/shared/constants/path.constant.ts]: API (but more suitable API_URL) should be defined in environment file

File: afisha-client/src/app/shared/services/event.service.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+@Injectable()
+export class EventService {
+
+  constructor(private http: HttpClient) {
+
+  }
+
+  getEvents(): Observable<IEvent[]>{
+    return this.http.get<IEvent[]>(PATH.API + 'events', {
+      headers: new HttpHeaders({
```
Comment (doslechescake) [inline afisha-client/src/app/shared/services/event.service.ts]: Firstly - If you use something that repeated in every request - this thing have to defined in interceptor.
Secondly - in your case header is redundant. Angular already sent all requests as 'application/json'

File: afisha-client/src/app/shared/services/event.service.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        'Accept': 'application/json'
+      })
+    });
+  }
+
+  getOneEvent(id: number): Observable<IEvent> {
+    return this.http.get<IEvent>(PATH.API + 'events/' + id, {
+      headers: new HttpHeaders({
```
Comment (doslechescake) [inline afisha-client/src/app/shared/services/event.service.ts]: The same as above

File: afisha-client/src/app/shared/services/event.service.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        'Accept' : 'application/json'
+      })
+    });
+  }
+
+  createEvent(event: IEvent): Observable<IEvent>{
+    return this.http.post<IEvent>(PATH.API + 'event', JSON.stringify(event), {
```
Comment (doslechescake) [inline afisha-client/src/app/shared/services/event.service.ts]: JSON.stringify(event) is redundant

File: afisha-client/src/app/shared/services/event.service.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      headers: new HttpHeaders({
+        'Content-Type' : 'application/json'
+      })
+    });
+  }
+
+  updateEvent(event: IEvent, id: string): Observable<void>{
+    return this.http.put<void>(PATH.API + 'event/' + id, JSON.stringify(event), {
```
Comment (doslechescake) [inline afisha-client/src/app/shared/services/event.service.ts]: Observable<void> as response type and this.http.put<void> - the same thing. Left only this.http.put<void>

File: afisha-server/server/api/controllers/eventController.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
+const Event = require('../models/eventModel');
+const _ = require('lodash');
```
Comment (doslechescake) [inline afisha-server/server/api/controllers/eventController.js]: Remove lodash

**Comment by vlasichok** [comment]
@doslechescake  Please review

**Pull request merged**